src/main/host/memory_manager.rs

```diff
@@ -2000,6 +2011,20 @@ mod export {
         0
     }
 
+    #[no_mangle]
+    pub unsafe extern "C" fn memorymanager_readString<'a>(
+        reader: *const MemoryReader<'a, u8>,
+        str: *mut libc::c_char,
+        strlen: libc::size_t,
+    ) -> libc::ssize_t {
+        let reader = &*reader;
+        let dst = std::slice::from_raw_parts_mut(str as *mut u8, strlen);
```

Comment:
I think it would be good to assert that `str` here isn't null.

Reply:
I started to add `debug_assert`s, but I think we can depend on Rust to do it for us. e.g. https://doc.rust-lang.org/src/core/slice/raw.rs.html#130

Reply:
I don't think debug asserts in the standard library run, for example this code doesn't panic: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=9fbb2471cca367c2624d2a04767e37e8

Reply:
Oh wow, that's surprising. I'll go ahead and add it then. I see I also missed a few other comments somehow; will make another pass

Reply:
Done